src/components/GithubRepositoryEntry.js

```diff
@@ -0,0 +1,174 @@
+import StyledInput from './StyledInput';
+import StyledButton from './StyledButton';
+import StyledCheckbox from './StyledCheckbox';
+import Link from './Link';
+
+const enhance = compose(
+  withState('state', 'setState', ({ errors = {} }) => ({ errors })),
+  withHandlers({
+    getFieldError: ({ state }) => name => state.errors[name],
+    onChange: ({ setState }) => event => {
+      event.stopPropagation();
+      const { target } = event;
+
+      setState(state => ({
+        ...state,
+        [target.name]: target.value,
+        errors: { ...state.errors, [target.name]: null },
+      }));
+    },
+    onInvalid: ({ setState }) => event => {
+      event.persist();
+      event.preventDefault();
+      setState(state => ({
+        ...state,
+        errors: { ...state.errors, [event.target.name]: event.target.validationMessage },
+      }));
+    },
+  }),
+  // follows composition of onChange && onInvalid to access them from props
+  withHandlers({
+    getFieldProps: ({ state, onChange, onInvalid }) => name => ({
+      defaultValue: state[name] || '',
+      fontSize: 'Paragraph',
+      lineHeight: 'Paragraph',
+      onChange,
+      onInvalid,
+      type: 'text',
+      width: 1,
+    }),
+  }),
+);
+
+const RepositoryEntry = enhance(
+  ({ getFieldError, getFieldProps, onCreateCollective, radio, value, checked, state, creatingCollective }) => {
+    const { type } = value.owner;
+    const repositoryTypeName = type === 'User' ? 'Personal Repo' : 'Organization Repo';
+
+    return (
+      <Fragment>
+        <Container display="flex" justifyContent="space-between" alignItems="flex-start">
+          <Container display="flex">
+            <Box as="span" mr={3}>
+              {radio}
+            </Box>
+            <Box as="span" mr={3}>
+              <Github size={40} color={colors.black[300]} />
+            </Box>
+            <Box as="span">
+              <P fontWeight={500} fontSize="1.4rem">
+                {value.full_name}
+              </P>
+              <P textTransform="uppercase" color="black.400" fontSize="1rem">
+                {repositoryTypeName}
+              </P>
+            </Box>
+          </Container>
+          <Container>
+            <Box>
+              <P fontWeight={300} fontSize="1.2rem">
+                {value.stargazers_count} <Star size={12} />
+              </P>
+            </Box>
+          </Container>
+        </Container>
+        <Container width={1} mx={3} my={2} px={2}>
+          {value.description && (
+            <P color="black.600" fontSize="1.2rem" fontWeight="400">
+              {value.description}
+            </P>
+          )}
+          {checked && (
+            <Container
+              as="form"
+              my={3}
+              method="POST"
+              onSubmit={event => {
+                event.preventDefault();
+                const data = pick(state, ['name', 'slug']);
+                data.githubHandle = value.full_name;
```

Comment:
If "Create a Collective for the Repository" is picked, `githubHandle` should be the long one from the repostiory (ie: `facebook/jest`).

If "Create a Collective for the Organization" is picked, `githubHandle` should be the one from the organization (ie: facebook).

Reply:
Bonus point: even if "Create a Collective for the Organization" is picked, we would still like to get the repository that was passed, and so we need it to add an extra property for it: `githubRepo`